### PR TITLE
Fix #226: Create raydata_scan_monitor_component

### PR DESCRIPTION
**Pull request by e-carlin**

### CODE REVIEW

**Comment by e-carlin** [comment]
The CI failure can be ignored (if it won't cause ci run errors downstream). It marked a print inside of a comment as problematic https://github.com/radiasoft/pykern/issues/307

File: rsconf/component/raydata_scan_monitor.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        )
+        self.install_access(mode="700", owner=jc.rsconf_db.run_u)
+        self.install_directory(z.db_d)
+        self.install_directory(z.intake_d)
+
+    def _unit_env(self, z):
+        return pkconfig.to_environ(
+            ["raydata.pkcli.*"],
+            values=z.pkmerge(
+                PKDict(raydata=PKDict(pkcli=PKDict(scan_monitor=PKDict(db_dir=z.db_d))))
```
Comment (robnagler) [inline rsconf/component/raydata_scan_monitor.py:51]: This obscures what's going on. You can say:

`to_environ(['a.b.*'], values={'a.b.c.d': 3})`

which produces

`{'A_B_C_D': '3'}`

You don't always need to use PKDict if you are going right to pkmerge (which doesn't care what type of dict is the arg).

File: rsconf/component/raydata_scan_monitor.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                z.db_d,
+                [
+                    z.intake_d,
+                    # POSIT: python -c "import databroker; print(databroker.catalog_search_path())"
+                    f"/home/{jc.rsconf_db.run_u}/.local/share/intake",
+                    "rw",
+                ],
+            ],
+        )
+        self.install_access(mode="700", owner=jc.rsconf_db.run_u)
```
Comment (robnagler) [inline rsconf/component/raydata_scan_monitor.py:43]: z.run_u avoids denormalization.

**Review by robnagler — APPROVED** [review]
Cosmetic suggestions

File: rsconf/component/raydata_scan_monitor.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            jc,
+            image=z.docker_image,
+            env=self._unit_env(z),
+            cmd="raydata scan_monitor",
+            volumes=[
+                z.db_d,
+                [
+                    z.intake_d,
+                    # POSIT: python -c "import databroker; print(databroker.catalog_search_path())"
+                    f"/home/{jc.rsconf_db.run_u}/.local/share/intake",
```
[the remark this answers is not in the document]
Comment (e-carlin) [reply rsconf/component/raydata_scan_monitor.py]: 8613e9a

File: rsconf/component/raydata_scan_monitor.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+_DB_SUBDIR = "db"
+
+
+class T(component.T):
+    def internal_build_compile(self):
+        self.buildt.require_component("docker")
+        jc, z = self.j2_ctx_init()
+        z.db_d = systemd.docker_unit_prepare(self, jc).join(_DB_SUBDIR)
+        z.intake_d = jc.systemd.run_d.join("intake")
```
[the remark this answers is not in the document]
Comment (e-carlin) [reply rsconf/component/raydata_scan_monitor.py]: 8613e9a

File: rsconf/component/raydata_scan_monitor.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    def internal_build_write(self):
+
```
[the remark this answers is not in the document]
Comment (e-carlin) [reply rsconf/component/raydata_scan_monitor.py]: 8613e9a

File: rsconf/component/raydata_scan_monitor.py (hunk 1 above)
Comment (e-carlin) [reply rsconf/component/raydata_scan_monitor.py]: Cool thanks 8613e9a

File: rsconf/component/raydata_scan_monitor.py (hunk 2 above)
Comment (e-carlin) [reply rsconf/component/raydata_scan_monitor.py]: 8613e9a

**Pull request merged**